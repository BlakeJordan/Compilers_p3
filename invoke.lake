int f(int arg1,bool arg2){
	int c;
	int d;
	bool a;
	c = 2;
	return c;
}
int z(bool x) {
	return "boob";
}
void g(){
	bool b;
	int a;
	int d;
	int @ w;
	d = 10;
	a = z(butt);
	b = true;
	a = 4;
	w = 12;
	if(b) {
		f(a, true);
	}
	else {
		g();
	}
	while(b) {
		f(a, b);
		b = false;
	}
	a = a + d * d;
	
}