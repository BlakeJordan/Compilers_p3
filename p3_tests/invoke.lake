int f(int arg1,bool arg2){
}
void g(){
	int a;
	f(a, true);
	g();
}
