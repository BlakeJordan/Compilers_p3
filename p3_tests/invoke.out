int f(int arg1, bool arg2) {
    int c;
    c(int) = 2;
    return c(int);
}
void g() {
    bool b;
    int a;
    int d;
    int @ f;
    d(int) = 10;
    b(bool) = true;
    a(int) = 4;
    if(b(bool)) {
        f(int)(a(int),true);
    }
    else {
        g(->void)();
    }
    while(b(bool)) {
        f(int)(a(int),b(bool));
        b(bool) = false;
    }
    a(int) = (a(int)+(d(int)*d(int)));
}
