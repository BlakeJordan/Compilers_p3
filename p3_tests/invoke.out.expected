int f(int arg1,bool arg2){
}

void g() {
    int a;
    f(int,bool->int)(a(int), true);
	g(->void)();
}
